orchestra/slack.py

```diff
@@ -50,9 +50,25 @@ def create_project_slack_group(project):
     return project.slack_group_id
 
 
+def _random_string():
+    return ''.join(
+        random.choice('abcdefghijklmnopqrstuvwxyz1234567890')
+        for idx in range(4))
+
+
 def _project_slack_group_name(project):
     """
-    Return a unique identifier for project slack groups; must fit into slack's
-    21 char limit for group names.
+    Return a unique and readable identifier for project slack groups.
+
+    Slack group names are capped at 21 characters in length.
     """
-    return base64.b64encode(uuid1().bytes)
+    name = None
+    # The human-readable portion of the name (16 characters) involves
+    # slugifying the project short description.
+    descriptor = slugify(project.short_description)[:16].strip('-')
+    while True:
+        # Add 4 characters of randomness (~1.68 million permutations).
+        name = '{}-{}'.format(descriptor, _random_string())
+        if not Project.objects.filter(slack_group_id=name).exists():
+            break
```

Comment:
Maybe we need to actually check for duplicates with the slack APIs? As in, what if we have groups left over from testing/before the last DB reset or created for other reasons?

And if it turns out not to be unique, what's the failure mode? Does it error? post messages to the existing group? Overwrite the existing group?